azurelinuxagent/common/osutil/default.py

```diff
@@ -1112,15 +1116,26 @@ def set_dhcp_hostname(self, hostname):
     def restart_if(self, ifname, retries=3, wait=5):
         retry_limit=retries+1
         for attempt in range(1, retry_limit):
-            return_code=shellutil.run("ifdown {0} && ifup {0}".format(ifname), expected_errors=[1] if attempt < retries else [])
-            if return_code == 0:
+            try:
+                shellutil.run_command(["ifdown", ifname])
+                shellutil.run_command(["ifup", ifname])
                 return
```

Comment:
Unnecessary return?

Reply:
I don't think so, because the try is in the for loop. This return exits the for loop early.